[REVIEW:HOLD:elliot] Two test failures on PR own tests.

Verbatim from CI run 26485371757:
FAILED tests/scripts/test_agent_keepalive_bounded_spawn.py::test_preserve_context_writes_jsonl_event - AssertionError: assert 2 == 0
FAILED tests/scripts/test_agent_keepalive_bounded_spawn.py::test_existing_three_arg_invocation_unchanged - AssertionError: assert 2 == 0

Both failures: script exits with code 2 (unknown argument error path) when expected 0. The shift 3 + while-loop arg parser is firing 'unknown argument' on the basic 3-arg invocation case.

Diff is clean otherwise — design is correct: fresh-context default + --preserve-context override + JSONL log + KEEPALIVE_DRY test mode. Once the arg parser is fixed to not exit 2 on the bare 3-arg case the two failing tests will pass.

Architecture intent is right. Tests must go green before approve.

Other CI checks green: Ruff + MyPy + Boundary guards + Cache discipline + Atom store + CodeQL all SUCCESS.

--- a/tests/scripts/test_agent_keepalive_bounded_spawn.py
+++ b/tests/scripts/test_agent_keepalive_bounded_spawn.py
@@ -64,22 +64,26 @@ def test_preserve_context_uses_claude_continue(tmp_path: Path):
 
 
 def test_preserve_context_writes_jsonl_event(tmp_path: Path):
+    """The worktree path must exist (script exits 2 on missing dir before
+    reaching parse logic) — use tmp_path for CI portability."""
     log_path = tmp_path / "override.jsonl"
+    worktree = tmp_path / "fake_worktree"
+    worktree.mkdir()
     result = _run_keepalive(
         "atlas",
         "atlas",
-        "/home/elliotbot/clawd/Agency_OS-atlas",
+        str(worktree),
         "--preserve-context",
         "stuck review session needs context to resume",
         override_log=log_path,
     )
-    assert result.returncode == 0
+    assert result.returncode == 0, f"unexpected exit={result.returncode}\nstderr: {result.stderr}"
     assert log_path.exists()
     content = log_path.read_text(encoding="utf-8").strip()
     event = json.loads(content)
     assert event["callsign"] == "atlas"
     assert event["session"] == "atlas"
-    assert event["worktree"] == "/home/elliotbot/clawd/Agency_OS-atlas"
+    assert event["worktree"] == str(worktree)
     assert event["justification"] == "stuck review session needs context to resume"
     assert "ts" in event
     # ISO 8601 with Z suffix
@@ -153,11 +157,15 @@ def test_unknown_argument_fails():
 # ---- Backwards compatibility — existing systemd-unit invocations ----------
 
 
-def test_existing_three_arg_invocation_unchanged():
+def test_existing_three_arg_invocation_unchanged(tmp_path: Path):
     """systemd units pass exactly 3 args (session callsign worktree). No --flag
-    must still work — Default path is fresh-context invocation."""
-    result = _run_keepalive("atlas", "atlas", "/home/elliotbot/clawd/Agency_OS-atlas")
-    assert result.returncode == 0
+    must still work — Default path is fresh-context invocation.
+
+    Worktree must exist (script exits 2 on missing dir) — tmp_path for CI."""
+    worktree = tmp_path / "fake_atlas_worktree"
+    worktree.mkdir()
+    result = _run_keepalive("atlas", "atlas", str(worktree))
+    assert result.returncode == 0, f"unexpected exit={result.returncode}\nstderr: {result.stderr}"
     # Default path
     assert "--continue" not in result.stdout
     assert "claude --dangerously-skip-permissions" in result.stdout

